tools/release_packages

```diff
@@ -0,0 +1,372 @@
+    print(crayons.green(message, bold=True))
+
+def parse_args():
+    usage = """
+Script to automate the PyPA release of the zulip, zulip_bots and
+zulip_botserver packages.
+
+To make a release, execute the following steps in order:
+
+1. Run ./tools/provision
+2. Activate the virtualenv created by tools/provision
+3. For example, to make a release for version 0.3.5, run the
+   following command:
+
+./tools/release_packages 0.3.5 --build --release --push origin \
+    --update-requirements "/home/username/zulip"
+
+The above command would accomplish the following (in order):
+
+1. Increment the __version__ in zulip/__init__.py,
+   zulip_bots/__init__.py and zulip_botserver/__init__.py.
+2. --build: Build sdists and universal wheels for all packages.
+   The sdists and wheels for a specific package are under
+   <package_name>/dist/.
+3. --release: Upload all dists under <package_name>/dist/* to
+   PyPA using twine.
+4. --update-requirements: Update the requirements/ in the main
+   zulip repo to install off of the newest version of the packages.
+   Also increments PROVISION_VERSION in the main repo.
+5. --push origin: Commit the changes produced in Step 1 and 4,
+   checkout a new branch named release-<version>, generate a commit
+   message, commit the changes and push the branch to origin.
+"""
+    parser = argparse.ArgumentParser(usage=usage)
+
+    parser.add_argument('release_version',
+                        help='The new version number of the packages.')
+
+    parser.add_argument('--cleanup', '-c',
+                        action='store_true',
+                        default=False,
+                        help='Remove build directories (dist/, build/, egg-info/, etc).')
+
+    parser.add_argument('--build', '-b',
+                        action='store_true',
+                        default=False,
+                        help=('Build sdists and wheels for all packages.'
+                              ' sdists and wheels are stored in <package_name>/dist/*.'))
+
+    parser.add_argument('--release', '-r',
+                        action='store_true',
+                        default=False,
+                        help='Upload the packages to PyPA using twine.')
+
+    parser.add_argument('--push',
+                        metavar='origin or upstream',
+                        default=None,
+                        help=('Commit and push a commit changing package versions'
+                              ' (can be either "origin" or "upstream"). If "origin'
+                              ' is specified, a new branch named release-<version> is'
+                              ' checked out before committing and pushing. If'
+                              ' "upstream" is supplied, then master is checked out'
+                              ' before committing and pushing. The process is the'
+                              ' same for changes made to both repos.'))
+
+    parser.add_argument('--update-requirements',
```

Comment:
Since it's really only the zulip repo we want to update, I'd rename this to `update-zulip-main-repo` or something.

Reply:
Sure, I'll do that. :)